cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -126,7 +132,8 @@ public synchronized ApiResponse registerSchema(ClusterSchemaRequest clusterSchem
             false,
             clusterSchemaRequest.getProtocol(),
             clusterSchemaRequest.getClusterIdentification(),
-            schemaCompatibility);
+            schemaCompatibility.getRight(),
```

Comment:
If this schemaCompatibility is coming from Global, then we should not register on the subject. I think we can add that one check.